### PR TITLE
AG-8489 - Fix PieSeries datum key change triggering incorrect animation.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-8489

One more PieSeries animation fix... 🤞

### CODE REVIEW

File: packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts [hunk 1]
```diff
@@ -60,25 +60,54 @@ function randomise() {
...
+    return newData;
+}
 
-    options.data = newData;
+function remove(position: 'start' | 'mid' | 'end' = 'end') {
+    options.data = removeData(position, options.data!);
+    AgCharts.update(chart, options);
+}
+
+function change(position: 'start' | 'mid' | 'end' = 'end') {
```
Comment (lsjroberts) [inline packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts]: `position` doesn't appear to be used here?

File: packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts [hunk 2]
```diff
@@ -60,25 +60,54 @@ function randomise() {
...
+function removeData(position: 'start' | 'mid' | 'end' | number = 'end', inData: any[]) {
+    const newData = [...(inData ?? [])];
+    const index =
+        position === 'start'
+            ? 0
+            : position === 'mid'
+            ? Math.floor(newData.length / 2)
+            : position === 'end'
+            ? -1
+            : position;
```
Comment (lsjroberts) [inline packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts]: tasty ternary, doesn't matter too much cos testing code, but just thought I'd mention it

**Review by lsjroberts — APPROVED** [review]

File: packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts]: Good spot - fixed.

File: packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-website/src/content/docs/pie-series-test/_examples/animation/main.ts]: Fixed.

**Pull request merged**